source/trainer.py

```diff
@@ -125,17 +135,36 @@ def tokenize(data):
     train_dataset=train_data_tokenized,
     eval_dataset=validate_data_tokenized,
 )
-
+if MODEL_RUN_IN != "gdrive":  # it does not work from gdrive
+    trainer.args.push_to_hub = True
+    trainer.args.hub_strategy = "end"
+    trainer.args.hub_model_id = HF_REPO
 """Train the model from scratch or from checkpoint"""
 if TRAIN_FROM_CHECKPOINT is not None:
-    result = trainer.train(f"{model_path}/{TRAIN_FROM_CHECKPOINT}")
+    trainer.args.num_train_epochs += ADDITIONAL_TRAIN_EPOCHS
+    result = trainer.train(TRAIN_FROM_CHECKPOINT)
 else:
     result = trainer.train()
 
 print("Training finished")
 print(result)
 
+
 """Save the tokenizer, latest status of trained model and push it to hugging face."""
 tokenizer.save_pretrained(model_path)
 model.save_pretrained(model_path)
-trainer.push_to_hub(f"model_{formattedtime}")
+
+"""wandb finish"""
+wandb.finish()
+
+"""Save the trainer state which contains the metrics to json"""
+trainer.state.save_to_json(f"{model_path}/trainer_state.json")
+
+"""Ploting the history of the training"""
+from trainer_utils import get_history, plot_history
+
+history = get_history(trainer)
+plot_history(history)
+
+if MODEL_RUN_IN != "gdrive":  # not sure it works from gdrive # TOCHECK
```

Comment:
I don't know why... it did not for me, and I read some threads about it reporting some problem there.. The thing is that it keeps the notebook running and open in a kinda stuck state, obviously with credits running. 
It would be nice to figure out how this works but it still does not.
My take is that we upload the model manually on HF for the moment. It is not much to do anyway.